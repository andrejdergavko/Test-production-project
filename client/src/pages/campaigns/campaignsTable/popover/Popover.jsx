import React, { useState } from "react";
import { StatefulPopover, PLACEMENT } from "baseui/popover";
import { useStyletron } from "styletron-react";
import Menu from "../menu";

function Popover() {
  const [isOpen, setIsOpen] = useState(false);
  const [css] = useStyletron();
  const icon = css({
    width: "17px",
  });
  const iconWrapper = css({
    display: "flex",
    width: "40px",
    height: "40px",
    alignItems: "center",
    justifyContent: "center",
  });
  const active = css({
    backgroundColor: "#eee",
  });

  return (
    <StatefulPopover
      onOpen={() => {
        setIsOpen(true);
      }}
      onClose={() => {
        setIsOpen(false);
      }}
      content={() => <Menu />}
      placement={PLACEMENT.auto}
      overrides={{
        Inner: {
          style: () => {
            return {
              backgroundColor: "#fff",
            };
          },
        },
      }}
    >
      <div className={`${iconWrapper} ${isOpen ? active : ""}`}>
        <img
          className={icon}
          src="https://img.icons8.com/material-rounded/24/000000/menu-2.png"
          alt="menu"
        />
      </div>
    </StatefulPopover>
  );
}

export default Popover;
