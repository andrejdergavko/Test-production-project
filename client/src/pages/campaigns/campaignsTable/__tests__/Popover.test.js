import React from "react";
import renderer from "react-test-renderer";

import Popover from "../popover";

describe("Popover component:", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Popover />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
