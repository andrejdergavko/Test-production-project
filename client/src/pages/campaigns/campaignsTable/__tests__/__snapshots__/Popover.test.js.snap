// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover component: renders correctly 1`] = `
<div
  aria-controls={null}
  aria-expanded="false"
  aria-haspopup="true"
  className=" "
  onClick={[Function]}
>
  <img
    alt="menu"
    className=""
    src="https://img.icons8.com/material-rounded/24/000000/menu-2.png"
  />
</div>
`;
